torch/onnx/_internal/exporter/_building.py

```diff
@@ -225,80 +316,128 @@ def _process_python_constants_and_sequences(
+    for name, arg in named_inputs.items():
+        param = signature.params_map[name]
+        assert isinstance(
+            param, _schemas.Parameter
+        ), f"Expected Parameter, got {type(param)}"
+
+        if not isinstance(arg, (tuple, list)):
+            continue
+
+        if len(arg) == 0:
+            # Skip empty sequences
+            continue
+
+        # 1. Sequence input of ir.Value
+        if _allowed_types_are_sequence_types(param.type_constraint.allowed_types):
+            # Turn the list into a Sequence node
+            # Constant op creation will be handled by the variadic case below when calling
+            # the SequenceConstruct op.
+            named_inputs[name] = opset.SequenceConstruct(*arg)
+            continue
+
+        # 2. Variadic inputs
+        # NOTE: Variadic operators like Max can be called with mixed ir.Value and Python constants
+        # like `Max(0, ir.Value())`
+        # We need to convert the Python constants to Constant nodes
+        if param.variadic:
+            if all(isinstance(val, ir.Value) for val in arg):
+                # Skip the variadic input if all values are ir.Value
+                continue
+
+            dtype = _determine_input_dtype(param, arg, type_binding)
+            new_args = []
+            for val in arg:
+                if isinstance(val, ir.Value):
+                    new_args.append(val)
+                else:
+                    constant_tensor = ir.tensor(value=val, dtype=dtype)  # type: ignore[arg-type]
+                    constant_value = opset.Constant(value=constant_tensor)
+                    new_args.append(constant_value)
+            named_inputs[name] = new_args
+            continue
+        else:
+            # 3. Concat the list as a single input
+            # E.g. [Value, 42] should be converted to op.Concat(Value, Constant(42))
+            # when the expected input type is INT64
+            # We assume this only happens for 1D cases
+            if all(isinstance(val, ir.Value) for val in arg):
+                named_inputs[name] = opset.Concat(*arg)
+                continue
+
+            dtype = _determine_input_dtype(param, arg, type_binding)
+            new_args = []
+            for val in arg:
+                if isinstance(val, ir.Value):
+                    new_args.append(val)
+                elif val is None:
+                    # Skip None values
+                    continue
+                elif isinstance(arg, (ir.Tensor, ir.TensorProtocol)):
+                    new_args.append(opset.Constant(value=val))
+                else:
+                    # Turn the Python constant into 1D tensor for the constant
+                    assert isinstance(
+                        val, (bool, int, float)
+                    ), f"Expected int or float, got {type(val)}"
+                    new_args.append(
+                        _get_or_create_constant(constant_farm, [arg], dtype, opset)  # type: ignore[arg-type]
```

Comment:
@justinchuby I was trying to debug an error that was hitting this line, and was wondering if `arg` here is supposed to be `val`? Changing that locally seemed to fix the problem for me, and it seems from the wider context of the function that `val` might be intended here, but I don't really understand enough to say for sure. It also seems like `arg` might need to be changed to `val` a few lines above in `elif isinstance(arg, (ir.Tensor, ir.TensorProtocol)):` for the same reason?

Reply:
Thanks for spotting this. I think you are right. If you can send a patch PR that would be much appreciated. cc @titaiwangms

Reply:
@JackDunnNZ Please try out nightly: https://github.com/pytorch/pytorch/blob/fea771dcce8508c5d71755da65d1b97d6da13475/torch/onnx/_internal/exporter/_building.py#L442

Reply:
Thanks, confirmed it's fixed on nightly, sorry for the noise!